sys/net/application_layer/nanocoap/nanocoap.c

```diff
@@ -46,17 +47,14 @@ static uint32_t _decode_uint(uint8_t *pkt_pos, unsigned nbytes);
  */
 int coap_parse(coap_pkt_t *pkt, uint8_t *buf, size_t len)
 {
-    uint8_t *urlpos = pkt->url;
     coap_hdr_t *hdr = (coap_hdr_t *)buf;
-
     pkt->hdr = hdr;
 
     uint8_t *pkt_pos = hdr->data;
     uint8_t *pkt_end = buf + len;
 
-    memset(pkt->url, '\0', NANOCOAP_URL_MAX);
```

Comment:
Is this change documented?  If not, may it cause nasty surprises to existing code?

Reply:
Or does the change on line #L119 override always the URL field?

Reply:
```coap->pkt``` does not exist anymore, unless gcoap is used. In that case, #L119 does overwrite the field.

Anyhow, I've added another null-termination for the case that the option is not present.

Reply:
Ah, you mean whether removing the field might cause nasty surprises? Very probably. :)